ui/admin/app/components/agent/Agent.tsx

```diff
@@ -41,46 +39,31 @@ export function Agent({ className, currentThreadId, onRefresh }: AgentProps) {
 		useAgent();
 
 	const [agentUpdates, setAgentUpdates] = useState(agent);
-	const [loadingAgentId, setLoadingAgentId] = useState("");
 
 	useEffect(() => {
-		setAgentUpdates((prev) => {
-			if (agent.id === prev.id) {
-				return {
-					...prev,
-					aliasAssigned: agent.aliasAssigned,
-				};
-			}
-
-			return agent;
-		});
-	}, [agent]);
-
-	const getLoadingAgent = useSWR(
-		AgentService.getAgentById.key(loadingAgentId),
-		({ agentId }) => AgentService.getAgentById(agentId),
-		{
-			revalidateOnFocus: false,
-			refreshInterval: 2000,
+		if (agent.aliasAssigned === undefined && agent.alias) {
+			const intervalId = setInterval(() => {
+				refreshAgent();
+				if (agent.aliasAssigned !== undefined) {
+					clearInterval(intervalId);
+				}
+			}, 1000);
+
+			return () => clearInterval(intervalId);
 		}
-	);
+	}, [agent, refreshAgent]);
 
 	useEffect(() => {
-		if (!loadingAgentId) return;
-
-		const { isLoading, data } = getLoadingAgent;
-		if (isLoading) return;
-
-		if (data?.aliasAssigned) {
-			setAgentUpdates((prev) => {
-				return {
-					...prev,
-					aliasAssigned: data.aliasAssigned,
-				};
-			});
-			setLoadingAgentId("");
-		}
-	}, [getLoadingAgent, loadingAgentId]);
+		setAgentUpdates((prev) => {
```

Comment:
This is a tricky case because when you update an agent, it would put agent into reconcilation state where it will wipe out aliasAssigned, but it doesn't indicate where alias needs reconcilation or not. So the workaround in UI is that it should never reset agent.aliasAssigned unless it is specifically trying to modify the alias. Otherwise every random update will cause `aliasAssigned` to be null and make alias url spinning.